agreement/voteAggregator.go

```diff
@@ -248,8 +248,12 @@ func voteFresh(proto protocol.ConsensusVersion, freshData freshnessData, vote un
 		return fmt.Errorf("filtered vote from bad round: player.Round=%v; vote.Round=%v", freshData.PlayerRound, vote.R.Round)
 	}
 
-	if freshData.PlayerRound+1 == vote.R.Round && (vote.R.Period > 0 || vote.R.Step > next) {
-		return fmt.Errorf("filtered future vote from bad period or step: player.Round=%v; vote.(Round,Period,Step)=(%v,%v,%v)", freshData.PlayerRound, vote.R.Round, vote.R.Period, vote.R.Step)
+	if freshData.PlayerRound+1 == vote.R.Round {
+		if (vote.R.Period > 0 || vote.R.Step > next) {
+			return fmt.Errorf("filtered future vote from bad period or step: player.Round=%v; vote.(Round,Period,Step)=(%v,%v,%v)", freshData.PlayerRound, vote.R.Round, vote.R.Period, vote.R.Step)
+		}
+		// pipeline votes from next round period 0
+		return nil
```

Comment:
I think according to the spec we also relay {late,redo,down} votes. We can ignore those here possibly, but then we should also amend the spec to fit this behavior.

I would call `voteStepFresh` here and drop `vote.R.Step > next` from the conditional above (with our step as 0) to match the rest of this function.

Reply:
The spec has a typo here as well. I will update the spec and call voteStepFresh.